docs/source/admin/traffic_ops/installation.rst

```diff
@@ -218,6 +218,8 @@ Upgrading Traffic Ops
 =====================
 To upgrade from older Traffic Ops versions, stop the service, use :manpage:`yum(8)` to upgrade to the latest available Traffic Ops package, and use the ``admin`` tool to perform the database upgrade.
 
+.. tip:: In order to upgrade to the latest version of Traffic Ops, please be sure that you have first upgraded to the latest available minor or patch version of your current release. For example, if your current Traffic Ops version is 3.0.0 and version 3.1.0 is available, you must first upgrade to 3.1.0 before proceeding to upgrade to 4.0.0. (Specifically, this means running all migrations, :file:`traffic_ops/app/db/seeds.sql`, and :file:`traffic_ops/app/db/patches.sql` for the latest of your current major version - which should be handled by the :program:`app/db/admin` tool).
```

Comment:
you might want to mention the migration id that's the latest prior to the collapse.  This could help ensure they really have upgraded appropriately.

Reply:
This tip reads as if its required to go to 3.1 (and possibly 3.0) first before going to 4.0. Is that a true statement based on your changes?

Reply:
not necessarily upgrade the full system,  but the db migrations must be run to bring your data in line with the new collapsed schema...

Reply:
Got it. So if I'm on TC2.2, I need to apply all the migrations to bring my DB schema up to 3.<latest> before running this latest migration?

Reply:
Done.

Reply:
@limited yes -- that's right